From fab1949589fd4569860e8cff98554dceabda6537 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Sun, 15 Mar 2015 00:10:17 +0100
Subject: [PATCH 03/11] Pass to python the proper path separator

Do not assume the python interpreter uses the ':' path separator.
---
 common.mk              | 4 ++--
 configure.ac           | 4 ++++
 tests/warn/Makefile.am | 2 +-
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/common.mk b/common.mk
index e26c637..59753ee 100644
--- a/common.mk
+++ b/common.mk
@@ -10,7 +10,7 @@ INTROSPECTION_SCANNER = \
     env PATH=".libs:$(PATH)" \
         LPATH=.libs \
         CC="$(CC)" \
-        PYTHONPATH=$(top_builddir):$(top_srcdir) \
+        PYTHONPATH="$(top_builddir)$(PYTHONPATHSEP)$(top_srcdir)" \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
         $(top_builddir)/g-ir-scanner
@@ -38,7 +38,7 @@ INTROSPECTION_COMPILER_ARGS = \
 INTROSPECTION_DOCTOOL = \
     env PATH=".libs:$(PATH)" \
         LPATH=.libs \
-        PYTHONPATH=$(top_builddir):$(top_srcdir) \
+        PYTHONPATH="$(top_builddir)$(PYTHONPATHSEP)$(top_srcdir)" \
         UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
         UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
         $(top_builddir)/g-ir-doc-tool
diff --git a/configure.ac b/configure.ac
index a6a272d..aeea00d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -276,6 +276,10 @@ AM_CHECK_PYTHON_HEADERS(, AC_MSG_ERROR([Python headers not found]))
 if test "x$os_win32" = "xyes"; then
   AM_CHECK_PYTHON_LIBS(, AC_MSG_ERROR([Python libs not found. Windows requires Python modules to be explicitly linked to libpython.]))
 fi
+AC_MSG_CHECKING(for the Python path separator)
+PYTHONPATHSEP=$($PYTHON -c 'import sys, os; sys.stdout.write(os.pathsep)')
+AC_SUBST(PYTHONPATHSEP)
+AC_MSG_RESULT($PYTHONPATHSEP)
 
 dnl Not enabled by default until 3.6 cycle when we can propose mako as
 dnl an external dependency
diff --git a/tests/warn/Makefile.am b/tests/warn/Makefile.am
index 5f47df0..38b6338 100644
--- a/tests/warn/Makefile.am
+++ b/tests/warn/Makefile.am
@@ -21,7 +21,7 @@ TESTS = \
 	unknown-parameter.h \
 	unresolved-type.h
 
-LOG_COMPILER = env PYTHONPATH=$(top_builddir):$(top_srcdir) TOP_BUILDDIR=$(top_builddir) UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
+LOG_COMPILER = env PYTHONPATH=$(top_builddir)$(PYTHONPATHSEP)$(top_srcdir) TOP_BUILDDIR=$(top_builddir) UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
 	$(PYTHON) $(srcdir)/warningtester.py
 
 EXTRA_DIST = warningtester.py common.h $(TESTS)
-- 
2.7.0

